lesson_8/alexandra_nechita/8thHomework_Python - Classes Inheritance.py

```diff
@@ -0,0 +1,114 @@
+from datetime import date
+
+class General_Patient (object):
+    
+    hospital = 'Regina Maria'
+    
+    def __init__(self, first_name, last_name, cnp, admission_date, insurance):
+        self.first_name = first_name
+        self.last_name = last_name
+        self.cnp = cnp
+        self.admission_date = admission_date
+        self.insurance = insurance
+        
+    def is_dischargeable (self, days_of_hospitalization):
+        if (date.today() - date(self.admission_date[0],self.admission_date[1], self.admission_date[2])).days == days_of_hospitalization:
+            return f'The patient {self.first_name} {self.last_name} is to be discharged today, {date.today()} \n'
+        elif (date.today() - date(self.admission_date[0],self.admission_date[1], self.admission_date[2])).days < days_of_hospitalization:
+            return f'The patient {self.first_name} {self.last_name} is to be discharged {days_of_hospitalization - (date.today() - date(self.admission_date[0],self.admission_date[1], self.admission_date[2])).days} day(s) from today \n'
+        else:
+            return f'The health situation of the patient {self.first_name} {self.last_name} requires extended hospitalization \n'
+        
+    def payment_status (self, cost_per_day, days_of_hospitalization):
+        if self.insurance:
+            return f'The patient {self.first_name} {self.last_name} has nothing to pay \n'
+        else:
+            return f'The patient {self.first_name} {self.last_name} has to pay {cost_per_day*days_of_hospitalization} ron \n'
+        
+    def __repr__(self):
+        class_name = type(self).__name__
+        return "Dev view: "'{}. Name: {} {}; CNP: {}; ID: {}; \n'.format(class_name, self.first_name, self.last_name, self.cnp, id(self))
+        
+    def __str__(self):
+        if self.insurance:
+            return f'This is the patient {self.first_name} {self.last_name}, with the CNP {self.cnp}. The patient has insurance. \n'
+        else:
+            return f'This is the patient {self.first_name} {self.last_name}, with the CNP {self.cnp}. The patient has no insurance. \n'
+    
+    @staticmethod
+    def hospital_motto():
+        return 'Good is not enough \n'
+
+class Diabetic_Patient (General_Patient):
+    
+    def __init__(self, first_name, last_name, cnp, admission_date, insurance, medication):
+        self.medication = medication
+        super().__init__(first_name, last_name, cnp, admission_date, insurance)
+    
+    def is_at_risk (self):
+        current_date = date.today()
+        cnp_as_list = [int(cifra) for cifra in str(self.cnp)]
+        if cnp_as_list[0] == 1 or cnp_as_list[0] == 2:

```

Comment:
aici ma gandesc ca vrei sa verifici daca cnp-ul este valid. Din cauza ca structura cnp-ului din Romania ii cunoscuta in principal de romani 😄 aici merge o metoda extra de genu 
```
def is_cnp_valid(self):
    return cnp_as_list[0] == 1 or cnp_as_list[0] == 2
```
sau 

```
def is_cnp_valid(self):
    # 1 and 5 -> male
    #  2 and 6 -> female
    return cnp_as_list[0] in [1,2,5,6]
```